code/modules/shuttle/supply.dm

```diff
@@ -67,10 +67,9 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 /obj/docking_port/mobile/supply/proc/check_blacklist(areaInstances)
 	for(var/place in areaInstances)
 		var/area/shuttle/shuttle_area = place
-		for(var/trf in shuttle_area)
-			var/turf/T = trf
-			for(var/a in T.GetAllContents())
-				if(is_type_in_typecache(a, GLOB.blacklisted_cargo_types) && !istype(a, /obj/docking_port))
+		for(var/turf/shuttle_turf as anything in shuttle_area)
+			for(var/item in shuttle_turf.GetAllContents())
+				if((is_type_in_typecache(item, GLOB.blacklisted_cargo_types) && !istype(item, /obj/docking_port)) || SSeigenstates.eigen_id[item]) //SSeigenstates is if it is linked to another object - block it
```

Comment:
This affects more than CentCom. What if I make an eigenstate locker, or whatever, on the emergency shuttle? Will it just stay in place?

Reply:
Also I think this should become a trait. Give it to `docking_port` as an `INNATE_TRAIT` or whatever, and apply it when something is splashed with the chemical, or however its done, with its own source that can be removed later on, if that's a feature.

Reply:
> 
> 
> This affects more than CentCom. What if I make an eigenstate locker, or whatever, on the emergency shuttle? Will it just stay in place?

Is the emergency shuttle a subtype of supply? How do living mobs get sent on the shuttle then? These items are also blacklisted
```dm
GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
		/mob/living,
		/obj/structure/blob,
		/obj/effect/rune,
		/obj/structure/spider/spiderling,
		/obj/item/disk/nuclear,
		/obj/machinery/nuclearbomb,
		/obj/item/beacon,
		/obj/narsie,
		/obj/tear_in_reality,
		/obj/machinery/teleport/station,
		/obj/machinery/teleport/hub,
		/obj/machinery/quantumpad,
		/obj/effect/mob_spawn,
		/obj/effect/hierophant,
		/obj/structure/receiving_pad,
		/obj/item/warp_cube,
		/obj/machinery/rnd/production, //print tracking beacons, send shuttle
		/obj/machinery/autolathe, //same
		/obj/projectile/beam/wormhole,
		/obj/effect/portal,
		/obj/item/shared_storage,
		/obj/structure/extraction_point,
		/obj/machinery/syndicatebomb,
		/obj/item/hilbertshotel,
		/obj/item/swapper,
		/obj/docking_port,
		/obj/machinery/launchpad,
		/obj/machinery/disposal,
		/obj/structure/disposalpipe,
		/obj/item/hilbertshotel,
		/obj/machinery/camera,
		/obj/item/gps,
		/obj/structure/checkoutmachine
	)))
	```

Reply:
>
>
> Also I think this should become a trait. Give it to docking_port as an INNATE_TRAIT or whatever, and apply it when something is splashed with the chemical, or however its done, with its own source that can be removed later on, if that's a feature.

Ah, I origianlly had it like that, but during review it was suggested to make the system seperate. I'll see if I can figure out traits.

Reply:
One thing I don't understand is why `/obj/docking_port` is seemingly checked twice.

Reply:
Probably just that the coder didn't know, you don't need an innate trait for it then.

Reply:
> Is the emergency shuttle a subtype of supply?

Nevermind on that bit then, didn't notice.

Reply:
Alright, done the requested changes!